binary-search-tree/node-js/inventory-management.js

```diff
@@ -19,15 +19,40 @@ class BinarySearchTree {
     }
 
     search(id) {
-        console.error("you must implement search");
+        // console.error("you must implement search");
+        return this.checkId(id, this.root);
+    }
+    // exampleSearch(id) {
+    //     if (this.root.val.id === id) {
+    //         return this.root;
+    //     } else {
+    //         this.checkId(id, this.root);
+    //     }
+    // }
+    checkId(id, node) {
+        if (node === null) {
+            return null;
+        } else if (id === node.val.id) {
+            return node.val;
+        } else if (id < node.val.id) {
+            return this.checkId(id, node.left);
+        } else {
+            return this.checkId(id, node.right);
+        }
     }
 
     updateInventory(id, inventoryCount) {
-        console.error("you must implement updateInventory");
+        // console.error("you must implement updateInventory");
+        const node = this.checkId(id, this.root);
+
+        if (node === null) return;
+        node.inventoryCount = inventoryCount;
+        return node.inventoryCount;
```

Comment:
I think `product` is a more appropriate name for this variable, otherwise this looks correct. The question is... is the unit test correct 😁